File: app/src/main/java/domon/cn/gankio/contract/CategoryContract.java
package domon.cn.gankio.contract;

import java.util.List;

import domon.cn.gankio.BasePresenter;
import domon.cn.gankio.BaseView;
import domon.cn.gankio.data.GankInfoData;

/**
 * Created by Domon on 16-10-28.
 */

public interface CategoryContract {
    interface View extends BaseView<Presenter> {
        void reqCategoryData(int type);

        void setCategoryDate(List<GankInfoData> gankInfoDatas);
    }

    interface Presenter extends BasePresenter {
        void reqCategoryData(int type, String index, String count);
    }
}

File: app/src/main/java/domon/cn/gankio/contract/CategoryContract.kt
package domon.cn.gankio.contract

import domon.cn.gankio.BasePresenter
import domon.cn.gankio.BaseView
import domon.cn.gankio.data.GankInfoData

/**
 * Created by Domon on 16-10-28.
 */

interface CategoryContract {
    interface View : BaseView<Presenter> {
        fun reqCategoryData(type: Int)

        fun setCategoryDate(gankInfoDatas: List<GankInfoData>)
    }

    interface Presenter : BasePresenter {
        fun reqCategoryData(type: Int, index: String, count: String)
    }
}
